fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/CascadesPlanner.java

```diff
@@ -206,10 +204,12 @@ public class CascadesPlanner implements QueryPlanner {
     @Nonnull
     private final RecordStoreState recordStoreState;
     @Nonnull
-    private final PlanningRuleSet ruleSet;
-    @Nonnull
     private Reference currentRoot;
     @Nonnull
+    private PlanContext planContext;
```

Comment:
Pulled up these two attributes from the nested task classes which all held references to these individually for no apparent reason. These two are immutable and stable during planning.